orchestra/static/orchestra/team-info/team-info-card.directive.es6.js

```diff
@@ -8,35 +8,58 @@ export default function teamInfoCard (orchestraApi) {
     template,
     restrict: 'E',
     scope: {
-      projectId: '='
+      taskAssignment: '='
     },
     controllerAs: 'teamInfoCard',
     bindToController: true,
     controller: ($scope) => {
       const teamInfoCard = $scope.teamInfoCard
-      orchestraApi.projectInformation(teamInfoCard.projectId)
-        .then(response => {
-          const {steps, tasks} = response.data
-          const humanSteps = new Set(steps.filter(step => step.is_human).map(step => step.slug))
-          teamInfoCard.steps = reduce(
-            Object.values(response.data.steps), (result, step) => {
-              result[step.slug] = step
-              return result
-            }, {})
-          teamInfoCard.assignments = []
-          for (let stepSlug of humanSteps.values()) {
-            const task = tasks[stepSlug]
-            if (task) {
-              teamInfoCard.assignments = teamInfoCard.assignments.concat(task.assignments.map(a => {
-                return {
-                  role: teamInfoCard.steps[stepSlug].name,
-                  worker: a.worker,
-                  recordedTime: moment.duration(a.recorded_work_time, 'seconds').roundMinute().humanizeUnits()
-                }
-              }))
+      teamInfoCard.projectId = teamInfoCard.taskAssignment.project.id
+      teamInfoCard.step = teamInfoCard.taskAssignment.step
+      teamInfoCard.isProjectAdmin = teamInfoCard.taskAssignment.is_project_admin
+
+      teamInfoCard.loadTeamInfo = () => {
+        orchestraApi.projectInformation(teamInfoCard.projectId)
+          .then(response => {
+            const {steps, tasks} = response.data
+            const humanSteps = new Set(steps.filter(step => step.is_human).map(step => step.slug))
+            teamInfoCard.steps = reduce(
+              Object.values(response.data.steps), (result, step) => {
+                result[step.slug] = step
+                return result
+              }, {})
+            teamInfoCard.assignments = []
+            for (let stepSlug of humanSteps.values()) {
+              const task = tasks[stepSlug]
+              if (task) {
+                teamInfoCard.assignments = teamInfoCard.assignments.concat(task.assignments.map(a => {
+                  return {
+                    stepSlug,
```

Comment:
stepSlug/status/task_id are used for the new task status/submit button functionality